src/Models/AdvectedPopulations/PISCES/phytoplankton.jl

```diff
@@ -0,0 +1,442 @@
+
+    return (1-δᴾ)*μᴾ*P - mᴾ*concentration_limitation(P, Kₘ)*P - sh*wᴾ*P^2 - gₚᶻ*Z - gₚᴹ*M #eq 1
+end
+
+#Diatom forcing
+@inline function (bgc::PISCES)(::Val{:D}, x, y, z, t, P, D, Z, M, Pᶜʰˡ, Dᶜʰˡ, Pᶠᵉ, Dᶠᵉ, Dˢⁱ, DOC, POC, GOC, SFe, BFe, PSi, NO₃, NH₄, PO₄, Fe, Si, CaCO₃, DIC, Alk, O₂, T, zₘₓₗ, zₑᵤ, Si̅, D_dust, Ω, κ, PAR, PAR₁, PAR₂, PAR₃)
+    #Parameters
+    δᴰ = bgc.exudation_of_DOC.D
+    mᴰ = bgc.phytoplankton_mortality_rate.D
+    Kₘ = bgc.half_saturation_const_for_mortality
+    αᴰ = bgc.initial_slope_of_PI_curve.D
+
+    τ₀ = bgc.background_shear
+    τₘₓₗ = bgc.mixed_layer_shear
+
+    sh = shear(z, zₘₓₗ, τ₀, τₘₓₗ)
+
+    #L_day
+    φ = bgc.latitude
+    φ = latitude(φ, y)
+
+
+    L_day = day_length(φ, t)
+    #Grazing
+    g_Dᶻ = grazing_Z(P, D, POC, T, bgc)[3]
+    g_Dᴹ = grazing_M(P, D, Z, POC, T, bgc)[3]
+
+    #Nutrient limitation
+    Lₗᵢₘᴰ = D_nutrient_limitation(D, PO₄, NO₃, NH₄, Si, Dᶜʰˡ, Dᶠᵉ, Si̅, bgc)[1]
+
+    #Also required
+    PARᴰ = D_PAR(PAR₁, PAR₂, PAR₃, bgc)
+    t_darkᴰ = bgc.mean_residence_time_of_phytoplankton_in_unlit_mixed_layer.D
+    μᴰ = phytoplankton_growth_rate(D, Dᶜʰˡ, PARᴰ, L_day, T, αᴰ, Lₗᵢₘᴰ, zₘₓₗ, zₑᵤ, κ, t_darkᴰ, bgc)
+    wᴰ = D_quadratic_mortality(D, PO₄, NO₃, NH₄, Si, Dᶜʰˡ, Dᶠᵉ, Si̅, bgc) #13
+
+    return (1-δᴰ)*μᴰ*D - mᴰ*concentration_limitation(D, Kₘ)*D - sh*wᴰ*D^2 - g_Dᶻ*Z - g_Dᴹ*M    #eq 9
+end
+
+#Forcing for chlorophyll biomass of nanophytoplankton
+@inline function (bgc::PISCES)(::Val{:Pᶜʰˡ}, x, y, z, t, P, D, Z, M, Pᶜʰˡ, Dᶜʰˡ, Pᶠᵉ, Dᶠᵉ, Dˢⁱ, DOC, POC, GOC, SFe, BFe, PSi, NO₃, NH₄, PO₄, Fe, Si, CaCO₃, DIC, Alk, O₂, T, zₘₓₗ, zₑᵤ, Si̅, D_dust, Ω, κ, PAR, PAR₁, PAR₂, PAR₃)
+   #Parameters
+    δᴾ = bgc.exudation_of_DOC.P
+    αᴾ = bgc.initial_slope_of_PI_curve.P
+    θₘᵢₙᶜʰˡ = bgc.min_ChlC_ratios_of_phytoplankton
+    θₘₐₓᶜʰˡᴾ = bgc.max_ChlC_ratios_of_phytoplankton.P
+    mᴾ = bgc.phytoplankton_mortality_rate.P
+    Kₘ = bgc.half_saturation_const_for_mortality
+    wᴾ = bgc.min_quadratic_mortality_of_phytoplankton
+
+    τ₀ = bgc.background_shear
+    τₘₓₗ = bgc.mixed_layer_shear
+
+    sh = shear(z, zₘₓₗ, τ₀, τₘₓₗ)
+
+    #L_day
+    φ = bgc.latitude
+    φ = latitude(φ, y)
+
+
+    L_day = day_length(φ, t)
+    #Grazing
+    gₚᶻ = grazing_Z(P, D, POC, T, bgc)[2]    
+    gₚᴹ = grazing_M(P, D, Z, POC, T, bgc)[2]
+
+    #Phytoplankton growth
+    t_darkᴾ = bgc.mean_residence_time_of_phytoplankton_in_unlit_mixed_layer.P
+    Lₗᵢₘᴾ= P_nutrient_limitation(P, PO₄, NO₃, NH₄, Pᶜʰˡ, Pᶠᵉ, bgc)[1]
+    PARᴾ = P_PAR(PAR₁, PAR₂, PAR₃, bgc)
+    μᴾ = phytoplankton_growth_rate(P, Pᶜʰˡ, PARᴾ, L_day, T, αᴾ, Lₗᵢₘᴾ, zₘₓₗ, zₑᵤ, κ, t_darkᴾ, bgc)
+
+    μ̌ᴾ = μᴾ / day_dependent_growth_rate(L_day) #15b
+    ρᴾᶜʰˡ = 144*μ̌ᴾ * P / (αᴾ* Pᶜʰˡ* ((PARᴾ)/(L_day + eps(0.0))) + eps(0.0)) #15a
+
+    return ((1-δᴾ)*(12*θₘᵢₙᶜʰˡ + (θₘₐₓᶜʰˡᴾ - θₘᵢₙᶜʰˡ)*ρᴾᶜʰˡ)*μᴾ*P - mᴾ*concentration_limitation(P, Kₘ)*Pᶜʰˡ 
+             - sh*wᴾ*P*Pᶜʰˡ - nutrient_quota(Pᶜʰˡ, P)*gₚᶻ*Z - nutrient_quota(Pᶜʰˡ, P)*gₚᴹ*M)  #14
```

Comment:
Is `- sh*wᴾ*P*Pᶜʰˡ` a typo in the paper, it doesn't have the right units if it is this right?

Reply:
@ciadht @hannahmw1